app/src/pages/project/DatasetSelectorPopoverContent.tsx

```diff
@@ -0,0 +1,140 @@
+import { graphql, useLazyLoadQuery, useRefetchableFragment } from "react-relay";
+import { css } from "@emotion/react";
+
+import {
+  Button,
+  Card,
+  Flex,
+  Icon,
+  Icons,
+  Item,
+  ListBox,
+  Text,
+  TextField,
+  View,
+} from "@arizeai/components";
+
+import { DatasetSelectorPopoverContent_datasets$key } from "./__generated__/DatasetSelectorPopoverContent_datasets.graphql";
+import { DatasetSelectorPopoverContentQuery } from "./__generated__/DatasetSelectorPopoverContentQuery.graphql";
+
+export type DatasetSelectorPopoverContentProps = {
+  onCreateNewDataset: () => void;
+  onDatasetSelected: (datasetId: string) => void;
+};
+export function DatasetSelectorPopoverContent(
+  props: DatasetSelectorPopoverContentProps
+) {
+  const { onCreateNewDataset, onDatasetSelected } = props;
+  const [search, setSearch] = useState<string>("");
+  const data = useLazyLoadQuery<DatasetSelectorPopoverContentQuery>(
+    graphql`
+      query DatasetSelectorPopoverContentQuery {
+        ...DatasetSelectorPopoverContent_datasets
+      }
+    `,
+    {},
+    { fetchPolicy: "network-only" }
+  );
+  return (
+    <Card
+      title="Add to Dataset"
+      variant="compact"
+      backgroundColor="light"
+      borderColor="light"
+      bodyStyle={{ padding: 0 }}
+      extra={
+        <Button variant="default" size="compact" onClick={onCreateNewDataset}>
+          New Dataset
+        </Button>
+      }
+    >
+      <View padding="size-100">
+        <TextField
+          type="search"
+          placeholder="Search datasets"
+          addonBefore={<Icon svg={<Icons.Search />} />}
+          onChange={(newSearch) => {
+            setSearch(newSearch);
+          }}
+        />
+      </View>
+      <View borderTopWidth="thin" borderColor="light">
+        <div
+          css={css`
+            height: 400px;
+            overflow-y: auto;
+          `}
+        >
+          <DatasetsListBox
+            query={data}
+            search={search}
+            onDatasetSelected={onDatasetSelected}
+          />
+        </div>
+      </View>
+    </Card>
+  );
+}
+
+function DatasetsListBox(props: {
+  search: string;
+  query: DatasetSelectorPopoverContent_datasets$key;
+  onDatasetSelected: (datasetId: string) => void;
+}) {
+  const { search, onDatasetSelected } = props;
+  const [data] = useRefetchableFragment(
+    graphql`
+      fragment DatasetSelectorPopoverContent_datasets on Query
+      @refetchable(queryName: "DatasetSelectorPopoverContentDatasetsQuery") {
+        datasets {
+          edges {
+            dataset: node {
+              id
+              name
+            }
+          }
+        }
+      }
+    `,
+    props.query
+  );
+  const datasets = useMemo(() => {
+    let datasets = data.datasets.edges.map((edge) => edge.dataset);
+    if (search) {
+      datasets = datasets.filter((dataset) =>
+        dataset.name.toLowerCase().includes(search.toLowerCase())
+      );
+    }
+    return datasets;
```

Comment:
Assuming this is the part we would like to make a LIKE query for. Do we have a ticket for that yet?